src/System.Memory/tests/Performance/Perf.Utf8Parser.cs

```diff
@@ -10,12 +11,984 @@ namespace System.Buffers.Text.Tests
+        }
+
+        [Benchmark]
+        #region Inline data
+        [InlineData("10737")] // standard parse
+        [InlineData("32767")] // max value
+        [InlineData("0")]
+        [InlineData("-32768")] // min value
+        [InlineData("2147")]
+        [InlineData("2")]
+        [InlineData("-21474")]
+        [InlineData("21474")]
+        [InlineData("-21")]
+        [InlineData("-2")]
+        [InlineData("214")]
+        [InlineData("2147")]
+        [InlineData("-2147")]
+        [InlineData("-48")]
+        [InlineData("48")]
+        [InlineData("483")]
+        [InlineData("21")]
+        [InlineData("-214")]
+        [InlineData("+21474")]
+        [InlineData("+21")]
+        [InlineData("+2")]
+        [InlineData("+214")]
+        [InlineData("+2147")]
+        [InlineData("+21475")]
+        [InlineData("+48")]
+        [InlineData("+483")]
+        [InlineData("+21437")]
+        [InlineData("000000000000000000001235")]
+        #endregion
+        private static void ByteSpanToInt16_BytesConsumed_Baseline(string text)
+        {
+            byte[] utf8ByteArray = Encoding.UTF8.GetBytes(text);
+            var utf8ByteSpan = new ReadOnlySpan<byte>(utf8ByteArray);
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        short.TryParse(text, out short value);
+                        TestHelpers.DoNotIgnore(value, 0);
+                    }
+                }
+            }
+        }
+
+        [Benchmark]
+        private static void ByteSpanToInt16_BytesConsumed_VariableLength_Baseline()
+        {
+            int textLength = s_Int16TextArray.Length;
+            byte[][] utf8ByteArray = (byte[][])Array.CreateInstance(typeof(byte[]), textLength);
+            for (var i = 0; i < textLength; i++)
+            {
+                utf8ByteArray[i] = Encoding.UTF8.GetBytes(s_Int16TextArray[i]);
+            }
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        short.TryParse(s_Int16TextArray[i % textLength], out short value);
+                        TestHelpers.DoNotIgnore(value, 0);
+                    }
+                }
+            }
+        }
+
+        [Benchmark]
+        [InlineData("True")]
+        [InlineData("False")]
+        private static void BaselineStringToBool(string text)
+        {
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
```

Comment:
The InnerIterationCount is set by the BenchmarkIterator (https://github.com/Microsoft/xunit-performance/blob/9d19c5b289d0ce791fc8f0dc05074657b1b314ec/src/xunit.performance.core/BenchmarkIterator.cs#L17).

Should it be set to a number?

Reply:
> I think it defaults to 1, should it be set to something else?

If the default is 1, we should remove the for loop. However, while we are using xunit.perf, I think we should keep the `private const int InnerCount = 100000;` to get meaningful results from the benchmark. Once we move this to BenchmarkDotNet, we can remove the inner count then. Otherwise, with inner count as 1, I would imagine the results will be inconsistent and noisy.